### PR TITLE
[bitnami/etcd] only add the service to advertised addresses when it is enabled

### PR DESCRIPTION
**Pull request by svenwiltink**
### Description of the change

Title is self explanatory

### Benefits

Removes a non existing service url from the advertised address list

### Applicable issues

fixes https://github.com/bitnami/charts/issues/18209

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution, please note it is needed to bump the chart version in the _Chart.yaml_ file

**Comment by svenwiltink** [comment]
@carrodher thanks for the reminder, forgot about that. Besides bumping the version, is the fix in the chart itself ok regarding the issue I opened or are there more changes required?

File: bitnami/etcd/templates/statefulset.yaml [hunk 1]
```diff
@@ -179,7 +179,7 @@ spec:
               {{- end }}
             {{- end }}
             - name: ETCD_ADVERTISE_CLIENT_URLS
-              value: "{{ $etcdClientProtocol }}://$(MY_POD_NAME).{{ $etcdHeadlessServiceName }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ .Values.containerPorts.client }},{{ $etcdClientProtocol }}://{{ $etcdFullname }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ coalesce .Values.service.ports.client .Values.service.port }}"
+              value: "{{ $etcdClientProtocol }}://$(MY_POD_NAME).{{ $etcdHeadlessServiceName }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ .Values.containerPorts.client }}{% if .Values.service.enabled %},{{ $etcdClientProtocol }}://{{ $etcdFullname }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ coalesce .Values.service.ports.client .Values.service.port }}{% endif %}"
```
Comment (FraPazGal) [inline bitnami/etcd/templates/statefulset.yaml:182]: I think you made a format typo here:

```suggestion
              value: "{{ $etcdClientProtocol }}://$(MY_POD_NAME).{{ $etcdHeadlessServiceName }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ .Values.containerPorts.client }}{{- if .Values.service.enabled }},{{ $etcdClientProtocol }}://{{ $etcdFullname }}.{{ .Release.Namespace }}.svc.{{ $clusterDomain }}:{{ coalesce .Values.service.ports.client .Values.service.port }}{{- end }}"
```

**Review by FraPazGal — COMMENTED** [review]
Hi @svenwiltink, I understand the logic behind removing the svc URL from `ETCD_ADVERTISE_CLIENT_URLS` when `service.enabled=false` but reading your use case in https://github.com/bitnami/charts/issues/18209 wouldn't you still have the same problem? 

If you are using the envar as a list of endpoints, the difference between having or not the svc URL when the service is disabled is the fact that you will have an endpoint that won't resolve or not have it altogether right? What I mean is that, having the service disabled, its URL advertised shouldn't affect you right?

**Comment by svenwiltink** [comment]
> Hi @svenwiltink, I understand the logic behind removing the svc URL from `ETCD_ADVERTISE_CLIENT_URLS` when `service.enabled=false` but reading your use case in #18209 wouldn't you still have the same problem?
> 
> If you are using the envar as a list of endpoints, the difference between having or not the svc URL when the service is disabled is the fact that you will have an endpoint that won't resolve or not have it altogether right? What I mean is that, having the service disabled, its URL advertised shouldn't affect you right?

The problem I'm describing in the linked issue is etcd-defrag using all the advertised addresses to perform a healthcheck. Even when the dns doesn't resolve the defrag tool will still try to perform a healthcheck, which in turn fails. Removing the non existing endpoint entirely fixes that particular problem.

**Review by FraPazGal — APPROVED** [review]
Thanks for the explanation @svenwiltink, LGTM!

**Pull request merged**